File: dasbod-front/lib/data/budget/budget-entities.ts
import { EntityData, Reference } from "@/lib/data/common"

export interface BudgetPlan extends EntityData {
  amountAtStart: number
  expectedIncome: number
  startDate: string
  endDate: string
}

export interface BudgetCategory extends EntityData {
  name: string
  amount: number
  budgetPlanRef: Reference
}

export interface BudgetTransaction extends EntityData {
  date: string
  categoryRef: Reference
  amount: number
  description: string
  status: BudgetTransactionStatus
}

export const budgetTransactionStatuses = [
  "PAID",
  "PROCESSING",
  "PLANNED",
] as const

export type BudgetTransactionStatus = typeof budgetTransactionStatuses

File: budgetin-back/budgetin-domain/src/main/kotlin/dev/gallon/domain/entities/BudgetEntities.kt
package dev.gallon.domain.entities

import kotlinx.datetime.LocalDate
import kotlinx.serialization.Serializable

@Serializable
data class BudgetPlan(
    val amountAtStart: Double,
    val expectedIncome: Double,
    val startDate: LocalDate,
    val endDate: LocalDate,
) : EntityData

@Serializable
data class BudgetCategory(
    val name: String,
    val amount: Double,
    val budgetPlanId: String,
) : EntityData

@Serializable
data class BudgetTransaction(
    val date: LocalDate,
    val categoryId: String,
    val amount: Double,
    val description: String,
    val status: BudgetTransactionStatus,
) : EntityData

enum class BudgetTransactionStatus {
    PAID,
    PROCESSING,
    PLANNED,
}
